fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/socialmedia/ChirpListAdapter.java

```diff
@@ -0,0 +1,90 @@
+package com.github.dedis.popstellar.ui.socialmedia;
+
+import static android.text.format.DateUtils.getRelativeTimeSpanString;
+
+import android.content.Context;
+import android.graphics.Color;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.BaseAdapter;
+import android.widget.ImageButton;
+import android.widget.TextView;
+
+import com.github.dedis.popstellar.R;
+import com.github.dedis.popstellar.model.objects.Chirp;
+import com.github.dedis.popstellar.model.objects.security.PublicKey;
+
+import java.util.List;
+
+public class ChirpListAdapter extends BaseAdapter {
+
+  private final SocialMediaViewModel socialMediaViewModel;
+  private List<Chirp> chirps;
+  private final LayoutInflater layoutInflater;
+
+  public ChirpListAdapter(
+      Context context, SocialMediaViewModel socialMediaViewModel, List<Chirp> chirps) {
+    this.socialMediaViewModel = socialMediaViewModel;
+    this.chirps = chirps;
+    layoutInflater = LayoutInflater.from(context);
+  }
+
+  public void replaceList(List<Chirp> chirps) {
+    this.chirps = chirps;
+    notifyDataSetChanged();
+  }
+
+  @Override
+  public int getCount() {
+    return chirps != null ? chirps.size() : 0;
+  }
+
+  @Override
+  public Chirp getItem(int position) {
+    return chirps.get(position);
+  }
+
+  @Override
+  public long getItemId(int position) {
+    return position;
+  }
+
+  @Override
+  public View getView(int position, View view, ViewGroup viewGroup) {
+    View chirpView = layoutInflater.inflate(R.layout.chirp_card, null);
```

Comment:
There are a code smell about using null as the root element and another about not reusing the view. Something like this could fix the code smells.
```suggestion
  public View getView(int position, View chirpView, ViewGroup viewGroup) {
    if (chirpView == null) {
      chirpView = layoutInflater.inflate(R.layout.chirp_card, viewGroup);
    }
```

Reply:
Putting viewGroup in inflate crashes the app on refresh